docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -123,6 +129,10 @@ class ComponentExample extends React.PureComponent<IComponentExampleProps, IComp
     ) {
       this.clearActiveState()
     }
+    const { themeName } = nextProps
+    if (this.state.themeName !== themeName) {
```

Comment:
a bit unsure - do we really need to store `themeName` as state?

Reply:
Ah, I got it now. This was added because we want to invoke renderSourceCode if the themeName prop is changed. Since we are invoking this method here, we need to have a way to get the new themeName, that's why I have it in the state as well. Any other idea about how can we achive this?

Reply:
actually, yes - but just to unblock the PR, we could make a notch, merge it and I will try to experiment on that further. To me `themeName` has clear immutable semantics (that is not example-specific - and this is a key difference from other ones, like `showRtl`, for example), so this is the reason I'd rather see it being introduced as prop